src/lib/ngc/ngcc-processor.ts

```diff
@@ -36,7 +43,7 @@ export class NgccProcessor {
     }
 
     const packageJsonPath = this.tryResolvePackage(moduleName, resolvedFileName);
-    if (!packageJsonPath) {
+    if (!packageJsonPath || this.ngccProcessingCache.hasProcessed(packageJsonPath)) {
```

Comment:
Don't you think that we can unify ngccProcessingCache / processedModules? More or less they serve the same purpose.

Reply:
The local `processedModules` still allows to avoid the `tryResolvePackage` call which does FS-based lookup, so it still has some benefit.

Reply:
I see. I didn't do that because `tryResolvePackage` uses both `moduleName` and `resolvedFileName`, where the latter depends on the resolution context as determined by the entry-point. So caching based on `moduleName` only directly in the `NgccProcessingCache` might not be fully equivalent.

Reply:
Simplified to just one cache per our discussion.